common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -144,6 +147,34 @@ public static long reallocateMemory(long address, long oldSize, long newSize) {
     return newMemory;
   }
 
+  /**
+   * Uses internal JDK APIs to allocate a DirectByteBuffer while ignoring the JVM's
+   * MaxDirectMemorySize limit.
```

Comment:
can you expand on this briefly to explain why we do this ourselves instead of calling `DirectByteBuffer` directly? It's cause the limit is too low (512m) by default right?